clang/lib/Sema/SemaExpr.cpp

```diff
@@ -14177,8 +14177,14 @@ QualType Sema::ValidateBoundsExprArgument(Expr *Arg) {
   // we do see them.
   Qualifiers ArgTypeQuals = ArgTypePointee.getQualifiers();
   ArgTypeQuals.removeCVRQualifiers();
-  assert(!ArgTypeQuals.hasQualifiers() &&
+
+  if (ArgTypeQuals.hasQualifiers()) {
+    if (!DisableSubstitionDiagnostics)
```

Comment:
When this method is called from PruneTemporaryBindings (as a result of calls to CheckUnaryOperator), there should be no assertion. Removing this change causes the Clang test Sema/address-unaligned.c to fail.